cocotb/handle.py

```diff
@@ -414,40 +386,61 @@ def _setcachedvalue(self, *args, **kwargs):
     def __le__(self, *args, **kwargs):
         raise ValueError("Not permissible to set values on a constant object")
 
-class NonConstantObject(NonHierarchyObject):
+class NonHierarchyIndexableObject(NonHierarchyObject):
     def __init__(self, handle):
         """
             Args:
                 _handle [integer] : vpi/vhpi handle to the simulator object
         """
         NonHierarchyObject.__init__(self, handle)
-        self._r_edge = _RisingEdge(self)
-        self._f_edge = _FallingEdge(self)
-        self._e_edge = _Edge(self)
-
-    def __hash__(self):
-        return self._handle
+        self._range_left  = simulator.get_range_left(self._handle)
```

Comment:
@stuarthodgson not sure I agree - this is useful information to have in Python, for example so that the iterator (which is entirely in Python) naturally operates in the expected direction.